File: programize/DS1/stack.py
import sys,os


class Stack():
	def __init__(self, **kwargs):
		self.size=kwargs['size']
		self.top = -1
		self.stack = []

	def isEmpty(self):
		return self.top < 0;

	def isFull(self):
		return self.top>=self.size

	def pop(self):
		if not self.isEmpty():
			element = self.stack[self.top]
			self.top -= 1
			return element
		else:
			print("stack is empty!!!")
			return None


	def push(self, x):
		if self.isFull():
			print("Stack is full!!!")
		else:
			self.top += 1
			self.stack.append(x)

	def peek(self):
		if not self.isEmpty():
			return self.stack[self.top]

def main():
	stack = Stack(size=10)
	for i in range(1,10):
		print(i)
		stack.push(i)
	
	for i in range(10):
		print(stack.pop())
	


if __name__ == '__main__':
	main()

File: programize/DS1/Stack.java
class Stack{
    int[] stack;
    int top = -1;
    int ssize;
    Stack(int ss){
        stack = new int[ss];
        ssize = ss;
    }

    boolean isEmpty(){
        return this.top<0;
    }

    boolean isFull(){
        return this.top>=this.ssize-1;
    }

    void push(int x){
        if(this.isFull())
            System.out.println("stack is full");
        else{
            this.top ++;
            this.stack[this.top] = x;
            System.out.println(x + " was added to stack successfully!");
        }
    }

    int pop(){
        if(this.isEmpty()){
            System.out.println("stack is empty");
            return -1;
        }
        else{
            int elem = this.stack[this.top];
            this.top -= 1;
            return elem;
            }
    }

    public static void main(String[] args){
        Stack mystack = new Stack(5);
        for(int i=0; i<7; i++)
            mystack.push(i);

        for(int i=0; i<mystack.ssize+2; i++){
            int item = mystack.pop();
            System.out.println(item);
        }

    }
}